From 576d79bfd897e737dafaacd919fdf917b312f75b Mon Sep 17 00:00:00 2001
From: Vincent Hou <sbhou@cn.ibm.com>
Date: Wed, 20 Nov 2013 04:34:38 -0500
Subject: [PATCH] Add search_opts into the method list() for VolumeTypeManager

Since the method base.ManagerWithFind.findall has been modified by
adding the search_opts for all tenants support, the parameter
search_opts should also be added to all the manager classes, which
can be called whithin this method. volumetypemanager does not have
this parameter for its method list(). Adding search_opts will resolve
this issue.

Closes-Bug: #1252665

Change-Id: I526500625f1b5483cb5d88ea15e6ac8485a66ae3

Conflicts:
	doc/source/index.rst
---
 cinderclient/v1/volume_types.py | 2 +-
 cinderclient/v2/volume_types.py | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/cinderclient/v1/volume_types.py b/cinderclient/v1/volume_types.py
index 12c4612..e8b2c39 100644
--- a/cinderclient/v1/volume_types.py
+++ b/cinderclient/v1/volume_types.py
@@ -80,7 +80,7 @@ class VolumeTypeManager(base.ManagerWithFind):
     """
     resource_class = VolumeType
 
-    def list(self):
+    def list(self, search_opts=None):
         """
         Get a list of all volume types.
 
diff --git a/cinderclient/v2/volume_types.py b/cinderclient/v2/volume_types.py
index bc382bd..3d76302 100644
--- a/cinderclient/v2/volume_types.py
+++ b/cinderclient/v2/volume_types.py
@@ -70,7 +70,7 @@ class VolumeTypeManager(base.ManagerWithFind):
     """Manage :class:`VolumeType` resources."""
     resource_class = VolumeType
 
-    def list(self):
+    def list(self, search_opts=None):
         """Get a list of all volume types.
 
         :rtype: list of :class:`VolumeType`.
